### PR TITLE
[DevTools] Validate Store operation invariants

### PR DESCRIPTION
**Pull request by hoxyq**
Builds on #37049 by validating Store operation invariants before mutation. Missing nodes, invalid element types, inconsistent parent-child relationships, and invalid reorder operations now emit and throw explicit errors instead of silently continuing with corrupted state.

Adds a canonical-render regression test for invalid child removal.

### CODE REVIEW

**Review by daltino — APPROVED** [review]
This PR improves `react-devtools-shared` by validating store operation invariants to ensure integrity during node removal, addressing potential edge cases. The added tests in `store-test.js` and expanded type handling in `store.js` are well-structured and align with React's contribution guidelines for maintaining reliability and preventing regressions.

File: packages/react-devtools-shared/src/devtools/store.js [hunk 1]
```diff
@@ -1059,10 +1129,6 @@ export default class Store extends EventEmitter<{
   ): Array<SuspenseTimelineStep> {
     const target: Array<SuspenseTimelineStep> = [];
     const focusedTransitionID = this._focusedTransition;
-    // $FlowFixMe[invalid-compare]
-    if (focusedTransitionID === null) {
-      return target;
-    }
```
Comment (eps1lon) [inline packages/react-devtools-shared/src/devtools/store.js:1065]: This should throw instead when `focusedTransitionID === 0` (i.e. the initial paint). We should always call `getSuspendableDocumentOrderSuspenseInitialPaint` when `focusedTransitionID === 0`

File: packages/react-devtools-shared/src/devtools/store.js [hunk 2]
```diff
@@ -140,6 +153,29 @@ function isNonZeroRect(rect: Rect) {
...
+    case ElementTypeSuspenseList:
+    case ElementTypeTracingMarker:
+    case ElementTypeVirtual:
+    case ElementTypeViewTransition:
+    case ElementTypeActivity:
+      return value;
+    default:
+      return null;
+  }
+}
```
Comment (eps1lon) [inline packages/react-devtools-shared/src/devtools/store.js:180]: We need to get a type error here if this isn't exhaustive for the known types.

File: packages/react-devtools-shared/src/devtools/store.js [hunk 3]
```diff
@@ -609,13 +645,11 @@ export default class Store extends EventEmitter<{
 
       if (root === undefined) {
         // We should never reach this. This is a bug in the backend renderer.
-        this._throwAndEmitError(
+        return this._throwAndEmitError(
```
Comment (eps1lon) [inline packages/react-devtools-shared/src/devtools/store.js:651]: aside: I don't get how Flow's `empty` is supposed to work. They say it's equivalent to TypeScript's `never` when you attempt to use `never` instead of `empty`. But `never` in TypeScript automatically marks every code after as unreachable which Flow doesn't do. Otherwise we wouldn't have to add a `return`. I guess Flow wants you to make it explicit with a `return`?

File: packages/react-devtools-shared/src/devtools/store.js (hunk 2 above)
Comment (hoxyq) [reply packages/react-devtools-shared/src/devtools/store.js:180]: Updated, thanks.

File: packages/react-devtools-shared/src/devtools/store.js (hunk 1 above)
Comment (hoxyq) [reply packages/react-devtools-shared/src/devtools/store.js:1065]: Updated, thanks

File: packages/react-devtools-shared/src/devtools/store.js (hunk 3 above)
Comment (hoxyq) [reply packages/react-devtools-shared/src/devtools/store.js:651]: > I guess Flow wants you to make it explicit with a return?

Seems like it. Without explicit `return` keyword, Flow doesn't mark the code below as unreachable and still type-checks it, which predictably fails. 

I don't know a real reason behind this, but I suppose it is just Flow lacking the code reachability checks that are present in TypeScript, while the idea behind `empty` and `never` is the same.

Comment (hoxyq) [reply packages/react-devtools-shared/src/devtools/store.js:651]: Here is [an example](https://flow.org/try/#1N4Igxg9gdgZglgcxALlAIwIZoKYBsD6uEEAztvhgE6UYCe+JADpdhgCYowa5kA0I2KAFcAtiRQAXSkOz9sAD0bZKcEYIncAdJCgkJ+RlQxjO3PgMXLV6rRJINaUDfMnTZIFhjATN0p9ZQQDCEJCBEMCTgwEH4hMnwhKABrKAgAdyh8OEywCLAAC3wANyo4LFxscWQuHnc2bCL8aFxaTQlaJXwWZlNa-nrmbFyJbDYAWnalXvMSJOwJAtox4VxcafdZ+cXl0RxKddioMAqqCY7sA5BEqCHKkiol7JKVDCdL69uSe8oliEZI6DcMYFDDZd66DAwbBjBDzEaUEhjMgSeFVGrmRJxUZjZhhOBkS4KJQqNROLQYL7KfR-ZQRCD7apmbAAX34kHqgQqEgABBhuQBebkABgA3AAdKASmCJbxwaDchRwCQACgAlMgFSJ-rRucAJdyDdyJPlKOluTc0tyAKLUelq8VQZkSqUygFQbnHUjYNUaooQOBsXX6w2KlWqh3Bg18wUARgdTslRyIZHtMRARWUJDlUECRSFmgAzAAmIuaIUgZlAA), `a = 1` statement is not marked as unreachable code, while the same line would be in TypeScript.

**Review by eps1lon — APPROVED** [review]

**Review by eps1lon — APPROVED** [review]
